ipn/ipnlocal/local.go

```diff
@@ -640,7 +643,9 @@ func (b *LocalBackend) linkChange(delta *netmon.ChangeDelta) {
 	hadPAC := b.prevIfState.HasPAC()
 	b.prevIfState = ifst
 	b.pauseOrResumeControlClientLocked()
-
+	if delta.Major {
```

Comment:
```suggestion
	if delta.Major && shouldAutoExitNode() {
```

Then remove the `&& shouldAutoExitNode` below in `setNetInfo` when checking `refresh` -- `refresh` will only be set to true if `shouldAutoExitNode()` is set.